app-api/src/main/java/org/openmrs/eip/app/CustomDatabaseOffsetBackingStore.java

```diff
@@ -0,0 +1,135 @@
+package org.openmrs.eip.app;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.MemoryOffsetBackingStore;
+import org.openmrs.eip.app.management.entity.DebeziumOffset;
+import org.openmrs.eip.app.management.repository.DebeziumOffsetRepository;
+import org.openmrs.eip.app.util.DebeziumOffsetUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.debezium.config.Instantiator;
+import io.debezium.embedded.EmbeddedEngine;
+
+@Component
```

Comment:
Why is this a spring bean? Does debezium allow you to hook in a backing store via spring?

Reply:
Resolved. Now we are using SyncContext to access the context content, like you suggested.